### PR TITLE
Fixes #20033 - load seed data just once

### PR DESCRIPTION
**Pull request by ares**

### CODE REVIEW

**Comment by ares** [comment]
To reproduce:
```
rake db:migrate:down VERSION=20170223114114
rake db:migrate db:seed
```

You will see
```
== 20170223114114 LockSeededTemplates: migrating ==============================
== 20170223114114 LockSeededTemplates: migrated (0.0041s) =====================

Seeding /home/ares/Projekty/Zdrojaky/foreman/db/seeds.d/02-permissions_list.rb
Seeding /home/ares/Projekty/Zdrojaky/foreman/db/seeds.d/02-roles_list.rb
...
Seeding /home/ares/Projekty/Zdrojaky/foreman/db/seeds.d/07-data.rb
/home/ares/Projekty/Zdrojaky/foreman/db/seeds.d/07-data.rb:16: warning: already initialized constant SEEDED_TEMPLATES
/home/ares/Projekty/Zdrojaky/foreman/db/seeds.d/07-data.rb:16: warning: previous definition of SEEDED_TEMPLATES was here
Seeding /home/ares/Projekty/Zdrojaky/foreman/db/seeds.d/07-provisioning_templates.rb
Seeding /home/ares/Projekty/Zdrojaky/foreman/db/seeds.d/08-data.rb
/home/ares/Projekty/Zdrojaky/foreman/db/seeds.d/08-data.rb:1: warning: already initialized constant SEEDED_PARTITION_TABLES
/home/ares/Projekty/Zdrojaky/foreman/db/seeds.d/08-data.rb:1: warning: previous definition of SEEDED_PARTITION_TABLES was here
Seeding /home/ares/Projekty/Zdrojaky/foreman/db/seeds.d/08-partition_tables.rb
```

**Review by dLobatog — APPROVED** [review]

**Review by dLobatog — CHANGES_REQUESTED** [review]
@ares This seems to cause a few problems on Katello & Foreman tests, mind to take a look? 

As far as I can see, this doesn't cause any change in behavior, it just removes the warning? Thanks!

**Comment by ares** [comment]
> As far as I can see, this doesn't cause any change in behavior

correct, it actually revealed an issue, the constant was being [modified](https://github.com/theforeman/foreman/blob/develop/db/seeds.d/07-provisioning_templates.rb#L6) by `input.delete(:source)` which was workarounded by loading the data file that redefined the constant just before seeding. I updated both seed files so that they don't touch input and only read data from it, it should fix tests

thanks and please take another look

**Review by dLobatog — APPROVED** [review]

File: db/seeds.d/07-provisioning_templates.rb [hunk 1]
```diff
@@ -17,13 +19,14 @@
...
       t = ProvisioningTemplate.create({
-        :snippet  => false,
-        :template => contents,
-        :locked => true
-      }.merge(input))
+                                        :template => contents,
+                                        :locked => true,
+                                        :default => true,
+                                        :name => input[:name],
+                                        :template_kind => input[:template_kind],
```
Comment (dLobatog) [inline db/seeds.d/07-provisioning_templates.rb:25]: Make sure the `template_kind` passed to in tests exists, either via FactoryGirl or fixtures. Tests failing mention `ActiveRecord::InvalidForeignKey: PG::ForeignKeyViolation: ERROR:  insert or update on table "templates" violates foreign key constraint "config_templates_template_kind_id_fk"... Key (template_kind_id)=(983253651) is not present in table "template_kinds".` which makes me think here's the issue

**Review by dLobatog — CHANGES_REQUESTED** [review]
@ares Thanks for the update, check out the template_kind in tests, some tests are still failing on Foreman/Katello.

**Comment by ares** [comment]
Seed test redefined the constant always with the new template kind which we started caching in a constant. I moved kinds seeding to a separate file but I'm afraid it won't help. I think it would be better if I move it to a separate class like we have for `RoleList`.

**Comment by ares** [comment]
Foreman test failures seems unrelated, [test upgrade] dies because "Slave went offline during the build"

**Review by dLobatog — APPROVED** [review]

**Comment by dLobatog** [comment]
Thanks @ares !

**Pull request merged**